podcasts/UpNextViewController+Swipe.swift

```diff
@@ -70,13 +70,21 @@ extension UpNextViewController: SwipeTableViewCellDelegate {
 
             if FeatureFlag.playlistsRebranding.enabled,
                let episode = DataManager.sharedManager.episodeInUpNextAt(index: indexPath.row + 1) as? Episode {
-                let shareAction = SwipeAction(style: .default, title: nil) { _, indexPath in
-                    NavigationManager.sharedManager.navigateTo(
-                        NavigationManager.manualPlaylistsChooserKey,
-                        data: [
-                            NavigationManager.manualPlaylistsChooserEpisodeKey: episode
-                        ]
-                    )
+                let shareAction = SwipeAction(style: .default, title: nil) { [weak self] _, indexPath in
+                    guard let self else { return }
+                    let presentModal: () -> Void = {
+                        NavigationManager.sharedManager.navigateTo(
+                            NavigationManager.manualPlaylistsChooserKey,
+                            data: [
+                                NavigationManager.manualPlaylistsChooserEpisodeKey: episode
+                            ]
+                        )
+                    }
+                    if self.presentingViewController is PlayerContainerViewController {
```

Comment:
@bjtitus as did for the paywall, in this particular case we need to dismiss the UpNext first and then present the new vc as we already reached the limit of presented views

Reply:
@bjtitus I just pushed a small change passing the root vc to present it from the UpNext in that case